libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp

```diff
@@ -986,10 +986,7 @@ void NavEKF3_core::selectHeightForFusion()
     bool rangeFinderDataIsFresh = (imuSampleTime_ms - rngValidMeaTime_ms < 500);
     const bool extNavDataIsFresh = (imuSampleTime_ms - extNavMeasTime_ms < 500);
     // select height source
-    if (extNavUsedForPos && (frontend->sources.getPosZSource() == AP_NavEKF_Source::SourceZ::EXTNAV)) {
```

Comment:
this isn't equivalent to the check #L1036 , the comments #L990 explain this. A merge of the two check with a || would have been a better IMHO

Reply:
Thank you @khancyr 

>the comments #L990 explain this

https://github.com/ArduPilot/ardupilot/blob/3a264e0a7699dc86dfd3d0815cfb3505dd8640ba/libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp#L990

I think this comment is a little outdated after EK3_SRCx_POSZ introduced. We can select ext nav as horizontal position  source and baro as vertical source now.